sql/parser/scan_test.go

```diff
@@ -267,9 +274,10 @@ func TestScanError(t *testing.T) {
 		sql string
 		err string
 	}{
-		{`1e`, "invalid floating point constant"},
-		{`1e-`, "invalid floating point constant"},
-		{`1e+`, "invalid floating point constant"},
+		{`1e`, "invalid floating point literal"},
+		{`1e-`, "invalid floating point literal"},
+		{`1e+`, "invalid floating point literal"},
+		{`0x`, "invalid hexadecimal literal"},
```

Comment:
Sure. Done. Did you take a look at both commits? I added a second commit which adds support for hex/octal/unicode string escapes.